WordPress/gutenberg
# [RNMobile] Add Reusable blocks to the inserter menu

`Gutenberg-mobile PR` -> https://github.com/wordpress-mobile/gutenberg-mobile/pull/2644



## Description
**This PR includes the changes from https://github.com/WordPress/gutenberg/pull/25265**

Add support to insert existing Reusable blocks from the editor in the mobile version. For this purpose the inserter menu now could show 2 tabs in case this kind of blocks have been previously created in the site.

The Reusable blocks tab will show all the blocks created of this type in the site.

## How has this been tested?

I tested this feature in the following 3 different scenarios:

<details>
<summary><strong>WordPress.com site</strong></summary>

Create a WordPress.com site or use an already created one.
- Create a post in the web version
- Add some blocks and create a Reusable block
- Open the app with metro running (Gutenberg-mobile)
- Add the WordPress.com site
- Create a new post or Edit a post
- Tap on the `+` icon to pop up the inserter menu
- Check that 2 tabs are shown, one (first one to be presented) with the blocks and another one with Reusable blocks
- Check that there're no Reusable blocks in the blocks tab
- Navigate to Reusable blocks tab
- Check that previously created Reusable blocks are shown

</details>

<details>
<summary><strong>Self-hosted site with Jetpack</strong></summary>

Create a self-hosted site with Jetpack (I used https://jurassic.ninja/create/) or use an already created one.
- Create a post in the web version
- Add some blocks and create a Reusable block
- Open the app with metro running (Gutenberg-mobile)
- Add the self-hosted site
- Create a new post or Edit a post
- Tap on the `+` icon to pop up the inserter menu
- Check that 2 tabs are shown, one (first one to be presented) with the blocks and another one with Reusable blocks
- Check that there're no Reusable blocks in the blocks tab
- Navigate to Reusable blocks tab
- Check that previously created Reusable blocks are shown

</details>

<details>
<summary><strong>Self-hosted site without Jetpack</strong></summary>

Create a self-hosted site without Jetpack (I created a local one using [this instructions](https://developer.wordpress.org/block-editor/contributors/develop/getting-started/#local-environment)) or use your own.
- Create a post in the web version
- Add some blocks and create a Reusable block
- Open the app with metro running (Gutenberg-mobile)
- Add the self-hosted site
- Create a new post or Edit a post
- Tap on the `+` icon to pop up the inserter menu
- Check that 2 tabs are shown, one (first one to be presented) with the blocks and another one with Reusable blocks
- Check that there're no Reusable blocks in the blocks tab
- Navigate to Reusable blocks tab
- Check that previously created Reusable blocks are shown

</details>

## Screenshots 

<details>
<summary>iOS Screenshots</summary>

## iPhone 8

![iPhone8-inserter-menu](https://user-images.githubusercontent.com/14905380/93384643-69e19c00-f865-11ea-9258-3fd126edcec9.gif)

Blocks tab | Reusable blocks tab | No reusable blocks created
------------ | ------------- | -------------
![Captura de pantalla 2020-09-16 a las 20 48 57](https://user-images.githubusercontent.com/14905380/93384525-3c94ee00-f865-11ea-8674-117ffcc99c66.png) | ![Captura de pantalla 2020-09-16 a las 20 49 02](https://user-images.githubusercontent.com/14905380/93384528-3dc61b00-f865-11ea-8a05-85d41235acca.png) | ![Captura de pantalla 2020-09-16 a las 21 19 06](https://user-images.githubusercontent.com/14905380/93384536-40c10b80-f865-11ea-9a58-9ce7a8701c08.png)

</details>

<details>
<summary>Android Screenshots</summary>

## Nexus 5

![nexus5-inserter-menu](https://user-images.githubusercontent.com/14905380/93383984-ba0c2e80-f864-11ea-9f0e-45bdcd9cddf3.gif)

Blocks tab | Reusable blocks tab | No reusable blocks created
------------ | ------------- | -------------
![Captura de pantalla 2020-09-16 a las 21 30 43](https://user-images.githubusercontent.com/14905380/93384052-d4dea300-f864-11ea-8ec6-865986d76651.png) | ![Captura de pantalla 2020-09-16 a las 21 32 26](https://user-images.githubusercontent.com/14905380/93384061-d740fd00-f864-11ea-9086-823c8761d0d5.png) | ![Captura de pantalla 2020-09-16 a las 21 46 59](https://user-images.githubusercontent.com/14905380/93385203-2f2c3380-f866-11ea-97d6-b47cbb14249a.png)

</details>

## Types of changes
New feature

## Checklist:
